tools/apps/src/bin/quiche-client.rs

```diff
@@ -83,430 +68,10 @@ fn main() {
-
-            trace!("written {}", write);
-        }
-
-        if conn.is_closed() {
-            info!("connection closed, {:?}", conn.stats());
-
-            if !conn.is_established() {
-                error!(
-                    "connection timed out after {:?}",
-                    app_data_start.elapsed(),
-                );
-
-                std::process::exit(HANDSHAKE_FAIL_STATUS);
-            }
-
-            if let Some(h_conn) = http_conn {
-                if h_conn.report_incomplete(&app_data_start) {
-                    std::process::exit(HTTP_FAIL_STATUS);
-                }
-            }
-
-            if let Some(si_conn) = siduck_conn {
-                si_conn.report_incomplete(&app_data_start);
-            }
-
-            break;
-        }
-    }
-}
-
-/// Application-specific arguments that compliment the `CommonArgs`.
-struct ClientArgs {
-    version: u32,
-    dump_response_path: Option<String>,
-    dump_json: bool,
-    urls: Vec<url::Url>,
-    reqs_cardinal: u64,
-    req_headers: Vec<String>,
-    no_verify: bool,
-    body: Option<Vec<u8>>,
-    method: String,
-    connect_to: Option<String>,
-}
-
-impl Args for ClientArgs {
-    fn with_docopt(docopt: &docopt::Docopt) -> Self {
-        let args = docopt.parse().unwrap_or_else(|e| e.exit());
-
-        let version = args.get_str("--wire-version");
-        let version = u32::from_str_radix(version, 16).unwrap();
-
-        let dump_response_path = if args.get_str("--dump-responses") != "" {
-            Some(args.get_str("--dump-responses").to_string())
-        } else {
-            None
-        };
-
-        let dump_json = args.get_bool("--dump-json");
-
-        // URLs (can be multiple).
-        let urls: Vec<url::Url> = args
-            .get_vec("URL")
-            .into_iter()
-            .map(|x| url::Url::parse(x).unwrap())
-            .collect();
-
-        // Request headers (can be multiple).
-        let req_headers = args
-            .get_vec("--header")
-            .into_iter()
-            .map(|x| x.to_string())
-            .collect();
-
-        let reqs_cardinal = args.get_str("--requests");
-        let reqs_cardinal = u64::from_str_radix(reqs_cardinal, 10).unwrap();
-
-        let no_verify = args.get_bool("--no-verify");
-
-        let body = if args.get_bool("--body") {
-            std::fs::read(args.get_str("--body")).ok()
-        } else {
-            None
-        };
-
-        let method = args.get_str("--method").to_string();
-
-        let connect_to = if args.get_bool("--connect-to") {
-            Some(args.get_str("--connect-to").to_string())
-        } else {
-            None
-        };
-
-        ClientArgs {
-            version,
-            dump_response_path,
-            dump_json,
-            urls,
-            req_headers,
-            reqs_cardinal,
-            no_verify,
-            body,
-            method,
-            connect_to,
-        }
+    match request(args, conn_args, stdout_sink) {
```

Comment:
`request` is a bit of an overly generic name, especially since the client is able to initiate many requests (depending in the args). Maybe something like `run` and to compliment `RunError`?

Reply:
I don't have any preferences here. @ghedo  is `run` fine by you?

Reply:
`make_requests`?

Reply:
I have modes that open QUIC connections and don't make requests e.g. siduck or a WebTransport branch that I have.

Reply:
How about `client::send`?

Reply:
How about "connect"? It communicates the intent while abstracting over some notions of sending or requesting

Reply:
fine by me

Reply:
updated the PR